kodilla-hibernate/src/main/java/com/kodilla/hibernate/invoice/Item.java

```diff
@@ -62,7 +63,7 @@ public int getProductQuantity()
     @Column(name = "itemValue")
     public BigDecimal getItemValue()
     {
-        return this.itemValue;
+        return getProductPrice().multiply(BigDecimal.valueOf(getProductQuantity()));
```

Comment:
tak to ja bym nie robił... czy to na pewno dobrze działa? Wrzuciłeś obliczenia w Java to gettera, który oznaczyłeś atrybutem @Column. Jeśli koniecznie chcesz użyć kolumny z wartością, która jest wyliczalna... rozważ użycie adnotacji @Formula:
https://thoughts-on-java.org/hibernate-tips-calculate-entity-attributes-formula/

Reply:
Działa dobrze, tj. w MySQL pojawiają się właściwe dane:
![image](https://user-images.githubusercontent.com/55031148/70378158-9901cf00-191d-11ea-8a65-2f301be9f5e9.png)
@Formula ogranicza wyliczalność do formuł SQL-owych. Generalnie koncepcja wstawienia pola wyliczalnego jest dość ryzykowana (sprzeczna z zasadą normalizacji bazy danych). Chciałem przetestować przy użyciu namedQueries (zakomentowany kod), co stanie się jak zmienię prze kontroler repozytoruim cenę z item1 (tego zapytania nie napisałem) a potem też przez kontroler sprawdzić czy pole itemValue ma właściwą wartość (to zapytanie jest). Nie wiem jednak dlaczego kod się nie chce skompilować.

Reply:
ok